packages/fiori-app-sub-generator/src/fiori-app-generator/end.ts

```diff
@@ -141,14 +142,25 @@ export async function runPostGenerationTasks(
     // Persist backend system connection information
     const hostEnv = getHostEnvironment();
 
-    if (service.backendSystem && hostEnv !== hostEnvironment.bas && service.backendSystem.newOrUpdated) {
+    if (
+        service.backendSystem &&
+        hostEnv !== hostEnvironment.bas &&
+        service.backendSystem.newOrUpdated &&
+        !service.backendSystem.temporaryCredentials
```

Comment:
Can we not remove the credentials from the PromptState.odataService.backendSystem object before returning (wherever we determine this) - so we dont need this extra logic in multiple places? I dont recall why we need this extra property `temporaryCredentials`?